src/3id/index.js

```diff
@@ -27,6 +33,38 @@ class ThreeId {
     this._keystore = keystore
   }
 
+  // TODO probably makes sense somwhere else, maybe keyring, though not part path, maybe utils
+  async newSymKey () {
+    return nacl.box.keyPair().secretKey
+  }
+
+  // TODO keep threads keys anywhere once loaded ?
+  // TODO where key funcs, maybe keyring if keys adde there, otherwise utils
+  async symEncrypt (symKey, message) {
+    let msg = utils.pad(message)
+
+    // TODO below redundant to whats in keyring
+    const nonce = randomNonce()
+    if (typeof msg === 'string') {
+      msg = nacl.util.decodeUTF8(msg)
+    }
+    const ciphertext = nacl.secretbox(msg, nonce, symKey)
+    return {
+      nonce: nacl.util.encodeBase64(nonce),
+      ciphertext: nacl.util.encodeBase64(ciphertext)
+    }
+  }
+
+  // TODO same as above, move to utils or somwhere
+  async symDecrypt (symKey, message) {
+    const ciphertext = nacl.util.decodeBase64(message.ciphertext)
+    const nonce = nacl.util.decodeBase64(message.nonce)
+    const cleartext = nacl.secretbox.open(ciphertext, nonce, symKey)
+    if (!cleartext) return null
+    const paddedMsg = nacl.util.encodeUTF8(cleartext)
+    return utils.unpad(paddedMsg)
+  }
+
```

Comment:
Don't these functions exist as helpers in the keyring class?

Reply:
yeah, just not exported, so need to be in some shared utils, since sym enc interface in keyring is only for sym key part of keyring